source/server/configuration_impl.cc

```diff
@@ -171,7 +172,25 @@ WatchdogImpl::WatchdogImpl(const envoy::config::bootstrap::v3::Watchdog& watchdo
   multikill_timeout_ =
       std::chrono::milliseconds(PROTOBUF_GET_MS_OR_DEFAULT(watchdog, multikill_timeout, 0));
   multikill_threshold_ = PROTOBUF_PERCENT_TO_DOUBLE_OR_DEFAULT(watchdog, multikill_threshold, 0.0);
-  actions_ = watchdog.actions();
+  auto actions = watchdog.actions();
+
+  // Add abort_action if killing is enabled.
+  envoy::watchdog::v3alpha::AbortActionConfig abort_config;
```

Comment:
nit: Move just before the 2 PackFrom calls below.

Although, would it work if you don't specify typed config since the proto in question has no fields set?

Reply:
It still works as we end up setting the type url which IIUC is necessary for finding the right factory for the typed config.